### ISSUE TITLE
Questions / comments after buffer refactor

### ISSUE THREAD

**Issue by steven-bellock** [DMTF/libspdm#1892]
@jyao1 

1. Why are `LIBSPDM_MAX_MESSAGE_VCA_BUFFER_SIZE`, `LIBSPDM_MAX_MESSAGE_C_BUFFER_SIZE`, etc, defined in https://github.com/DMTF/libspdm/blob/main/include/library/spdm_lib_config.h and not https://github.com/DMTF/libspdm/blob/main/include/internal/libspdm_lib_config.h? When would the Integrator want to override these values?
2. In addition, it seems like these value are only used if `LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT==1` so they should be gated by that macro.
3. `LIBSPDM_MAX_SENDER_RECEIVER_BUFFER_SIZE` should be moved to `unit_test/` since it's only used there.
4. There still seems to be a disconnect between the `libspdm_device_acquire_*` functions and these macros. In particular the `libspdm_device_acquire_` functions return `max_msg_size` to the caller. But if the size of the buffer is known at compile-time, why return the size at run-time? Note that the `libspdm_device_acquire_*` functions are registered at run-time and allow the Integrator to customize their `spdm_context`s based on the transport. They may have one context for use over I2C whose send/receive buffer is 2 KiB, and another context for use over PCIE whose send/receive buffer is 2 MiB. Can libspdm support this scenario with the current macros?

There are other issues but we can start here.

**Comment by jyao1** [comment]
1. Currently, `LIBSPDM_MAX_MESSAGE_VCA_BUFFER_SIZE` will be used by integrator to input VCA buffer. The integrator may override them because we are using `LIBSPDM_MAX_HASH_SIZE`, `LIBSPDM_MAX_ASYM_KEY_SIZE` and `LIBSPDM_MAX_DHE_KEY_SIZE` in `spdm_crypt_lib.h`. Maybe an enhancement is to redefine those CRYPTO size macro according to supported crypto algorithm. Then we can move them to internal.
2. That is good point. Agree.
3. The integrator may also use them, such as spdm-emu.
4. That is good example. I think we can use `LIBSPDM_DATA_CAPABILITY_DATA_TRANSFER_SIZE`, `LIBSPDM_DATA_CAPABILITY_MAX_SPDM_MSG_SIZE` for runtime. But I think we need add `LIBSPDM_DATA_SENDER_DATA_TRANSFER_SIZE`.

**Comment by jyao1** [comment]
All comments mentioned in this issue are resolved. Close this one.

If there is new comment, please file new issue.
